STM32 :  add  STM32C071 series

djiatsaf-st opened the pull request:
- Add soc and kconfig
- Provide support dtsi files
- Add board support for nucleo_c071rb
- Run samples like helloworld, blinky and button

djiatsaf-st committed:
soc: st: stm32: add soc for stm32c071

select soc for stm32c071 and irq configuration

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
dts: arm: st: add stm32c071 dtsi files

- provide support for the STM32C071 serie
- add stm32g0-flash-controller compatible on flash node
to fix CI issue on undefined reference to
`flash_stm32_page_layout'

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
boards: st: add support for nucleo_c071rb board

- add devicetree for nucleo_c071rb
- add board configurations files
- add documentation

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

nordicjm commented on boards/st/nucleo_c071rb/doc/index.rst:
```diff
@@ -0,0 +1,162 @@
+Supported Features
+==================
+
+The Zephyr nucleo_c071rb board configuration supports the following hardware features:
```
```suggestion
The Zephyr ``nucleo_c071rb`` board supports the following hardware features:
```

nordicjm commented on boards/st/nucleo_c071rb/nucleo_c071rb_defconfig:
```diff
@@ -0,0 +1,12 @@
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+# enable console
```
```suggestion
# Enable console
```

nordicjm commented on boards/st/nucleo_c071rb/doc/img/nucleo_c071rb.jpg:
Remove background (if possible), reduce width/height to remove black space, convert to webp, then put through https://tinypng.com/

FRASTM commented on boards/st/nucleo_c071rb/Kconfig.nucleo_c071rb:
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2023 Benjamin Björnsson <benjamin.bjornsson@gmail.com>
```
Copyright (c) 2024 etc

FRASTM commented on boards/st/nucleo_c071rb/arduino_r3_connector.dtsi:
```diff
@@ -0,0 +1,40 @@
+		gpio-map = <0 0 &gpioa 0 0>,	/* A0 */
+			   <1 0 &gpioa 1 0>,	/* A1 */
+			   <2 0 &gpioa 4 0>,	/* A2 */
+			   <3 0 &gpiob 1 0>,	/* A3 */
```
GPIO PB0 on A3, etc
see  the schematics

FRASTM commented on boards/st/nucleo_c071rb/doc/index.rst:
```diff
@@ -0,0 +1,162 @@
+----------------------------------
+
+- I2C1 SCL/SDA : PB8/PB9 (Arduino I2C)
+- LD4       : PA5
```
LD1 and LD4 on PC9

erwango commented on soc/st/stm32/stm32c0x/Kconfig.defconfig.stm32c071xx:
```diff
@@ -0,0 +1,11 @@
+# STMicroelectronics STM32C071xx MCU
+
+# Copyright (c) STMicroelectronics.
```
```suggestion
# Copyright (c) 2024 STMicroelectronics.
```

erwango commented on soc/st/stm32/stm32c0x/Kconfig.soc:
```diff
@@ -10,6 +10,10 @@ config SOC_SERIES_STM32C0X
 config SOC_SERIES
 	default "stm32c0x" if SOC_SERIES_STM32C0X
 
+config SOC_STM32C071XX
+	bool
+	select SOC_SERIES_STM32C0X
```
Please sort numerically

erwango commented on dts/arm/st/c0/stm32c071.dtsi:
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 Benjamin Björnsson <benjamin.bjornsson@gmail.com>
```
Copyright. Apply everywhere

erwango commented on boards/st/nucleo_c071rb/board.cmake:
```diff
@@ -0,0 +1,7 @@
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
+
+# keep first
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
```
Please add openocd to enable debug

erwango commented on boards/st/nucleo_c071rb/nucleo_c071rb.dts:
```diff
@@ -0,0 +1,154 @@
+	status = "okay";
+};
+
+&adc1 {
+	pinctrl-0 = <&adc1_in0_pa0 &adc1_in1_pa1 &adc1_in4_pa4>;
+	pinctrl-names = "default";
+	st,adc-clock-source = <SYNC>;
+	st,adc-prescaler = <4>;
+	status = "okay";
+	vref-mv = <3300>;
+};
+
+&die_temp {
+	status = "okay";
+};
+
+&vref {
+	status = "okay";
+};
```
Did you test all those ?

djiatsaf-st replied:
Yes, I did the tests on ADC and Dietemp

erwango reviewed (changes requested):
Otherwise LGTM

djiatsaf-st committed:
west.yml: update hal_stm32 to the latest version

It introduces *-pinctrl.dtsi files for the C071XX
STM32 series, which will help to fix a CI problem
encountered.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

nordicjm commented on boards/st/nucleo_c071rb/doc/index.rst:
```diff
@@ -0,0 +1,163 @@
+Supported Features
+==================
+
+The Zephyr ``nucleo_c071rb`` board configuration supports the following hardware features:
```
-configuration

nordicjm commented on boards/st/nucleo_c071rb/doc/index.rst:
```diff
@@ -0,0 +1,163 @@
+
+Nucleo C071RB board includes an ST-LINK/V2-1 embedded debug tool interface.
+
+Applications for the ``nucleo_c071rb`` board configuration can be built and
```
-configuration

gautierg-st commented on boards/st/nucleo_c071rb/nucleo_c071rb.dts:
```diff
@@ -0,0 +1,154 @@
+&adc1 {
+	pinctrl-0 = <&adc1_in0_pa0 &adc1_in1_pa1 &adc1_in4_pa4>;
+	pinctrl-names = "default";
+	st,adc-clock-source = <SYNC>;
```
```suggestion
	st,adc-clock-source = "SYNC";
```
Sorry for that, it's my fault

erwango commented:
@nordicjm PTAL
